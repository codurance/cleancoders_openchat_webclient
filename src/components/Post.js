import React from 'react'

const Post = ({ post }) => {
  return (
    <div className="row justify-content-md-center bg-light border rounded my-2 py-2">
      <div className="col-sm-8">
        {post.text}
      </div>
      <div className="col-sm-4 text-right">
        <span className="font-italic">Posted on {post.date} at {post.time}</span>
      </div>
    </div>
  )
}

export default Post
