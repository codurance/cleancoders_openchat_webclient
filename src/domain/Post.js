export default class Post {
  constructor({ id, userId, text, date, time }) {
    this.id = id
    this.userId = userId
    this.text = text
    this.date = date
    this.time = time
  }
}
