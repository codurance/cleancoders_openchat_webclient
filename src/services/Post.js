import axios from 'axios'
import Post from 'domain/Post'

const parse = data => new Post(
  {
    id: data.postId,
    userId: data.userId,
    text: data.text,
    dateTime: new Date(data.dateTime)
  })

class PostService {
  async createPostByUser (userId, text) {
    const request = {
      text : text
    }
    const response = await axios.post(`${process.env.REACT_APP_API_URL}user/${userId}/posts`,
      JSON.stringify(request) )

    return parse(response.data)
  }

  async getPostsOfUser(userId) {
    const response = await axios.get(`${process.env.REACT_APP_API_URL}user/${userId}/timeline`)

    return response.data.map( post => parse(post) )
  }

  async getWallOfUser (userId) {
    const response = await axios.get(`${process.env.REACT_APP_API_URL}user/${userId}/wall`)

    return response.data.map( post => parse(post) )
  }
}

export default new PostService()
