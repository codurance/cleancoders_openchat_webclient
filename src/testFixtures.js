import Post from 'domain/Post'
import User from 'domain/User'

export const aPost = new Post({
  id: 'an Id',
  userId: 'a user Id',
  text: 'Some text',
  date: '18/02/2018',
  time: '11:12:13'
})

export const anotherPost = new Post({
  id: 'another Id',
  userId: 'another user Id',
  text: 'Some other text',
  date: '19/02/2017',
  time: '13:14:15'
})

export const somePosts = [
  aPost,
  anotherPost
]

export const aUser = new User({
  id: 'some id',
  name: 'some name',
  about: 'about him'
})

export const anotherUser = new User({
  id: 'another id',
  name: 'another name',
  about: 'some another about text'
})

export const someUsers = [
  aUser,
  anotherUser
]
