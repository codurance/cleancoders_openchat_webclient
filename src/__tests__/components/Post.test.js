import React from 'react'
import { shallow } from 'enzyme'
import { Link } from 'react-router-dom'
import Post from 'components/Post'
import { aPost, aUser } from 'testFixtures'

const router = createMockRouter()
const context = { router }

describe('Post', () => {
  it('displays link and data', () => {
    const wrapper = shallow(<Post post={aPost} user={aUser} />, { context })
    const link = wrapper.find(Link)

    expect(link).toHaveLength(1)
    expect(link.prop('to')).toEqual(`/wall/${aUser.id}`)
    expect(wrapper.text().includes(aPost.text)).toBe(true)
    expect(wrapper.text().includes(aPost.date)).toBe(true)
    expect(wrapper.text().includes(aPost.time)).toBe(true)
  })
})
