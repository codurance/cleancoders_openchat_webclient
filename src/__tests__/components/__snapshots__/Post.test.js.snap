// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders 1`] = `

<div class="row justify-content-md-center bg-light border rounded my-2 py-2">
  <div class="col-sm-8">
    Some text
  </div>
  <div class="col-sm-4 text-right">
    <span class="font-italic">
      Posted on 18/02/2018 at 11:12:13
    </span>
  </div>
</div>

`;
